pkg/scanners/terraform/scanner_test.go

```diff
@@ -66,6 +66,24 @@ resource "something" "else" {}
 
 }
 
+func Test_TrivyOptionWithAlternativeIDProvider(t *testing.T) {
+	reg := rules.Register(alwaysFailRule, nil)
+	defer rules.Deregister(reg)
+
+	options := []options.ScannerOption{
+		ScannerWithAlternativeIDProvider(func(s string) []string {
+			return []string{"something", "altid", "blah"}
+		}),
+	}
+	results := scanWithOptions(t, `
+//trivy:ignore:altid
+resource "something" "else" {}
+`, options...)
+	require.Len(t, results.GetFailed(), 0)
+	require.Len(t, results.GetIgnored(), 1)
+
+}
+
```

Comment:
As most of the other inputs are the same, can we make this into a table driven test instead? So something like:

testCases := []struct{
name string
inputOptions string}{
{ name: "tfsec", inputOptions: "..."}}

then you can iterate over them as inputs:

for _, tc := range testCases {
t.Run(...)
}